subprojects/packagefiles/uchardet/meson.build

```diff
@@ -0,0 +1,30 @@
+project(
+  'uchardet',
+  ['c', 'cpp'],
+  version: '0.0.7',
+  license: ['MPL-1.1', 'GPL-2.0-or-later', '  LGPL-2.1-or-later'],
+  meson_version: '>=0.53.0',
+  default_options: ['cpp_std=c++14'],
+)
+
+base_cpp_args = []
+
+cxx = meson.get_compiler('cpp')
+host_cpu_family = host_machine.cpu_family()
+
+use_sse2 = not get_option('use_sse2').disabled()
+
+if host_cpu_family == 'x86_64' or host_cpu_family == 'x86'
```

Comment:
Now that I look at this again, you can probably get rid of this if. has_arguments() already checks if it can be used.

Reply:
I think its worth preserving this.

`-ffloat-store` seems to be an optimization that only makes sense on x86 and x86_64 where the float register gives excess precision when the precide definition of IEEE floats is required.